### PR TITLE
[Feature] Implement polypynet (#341)

### PR DESCRIPTION
**Pull request by AntoninLoubiere**
# Implement polypynet from #341 

I've implemented functions from ppn_all, all functions from ppn_nw are remaining to be implement and will be implemented in few days.

### CODE REVIEW

File: python/port/mod/ppn/modppn.h [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#include <py/obj.h>
+
+const int screenWidth = 320;
```
Comment (RedGl0w) [inline python/port/mod/ppn/modppn.h:3]: Use `#define` instead (if not reused)

File: python/port/mod/ppn/modppn_table.c [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(modppn_colLst2Int_obj, modppn_colLst2Int);
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(modppn_colInt2Lst_obj, modppn_colInt2Lst);
+
+STATIC const mp_rom_map_elem_t modppn_module_globals_table[] = {
+  { MP_ROM_QSTR(MP_QSTR___name__), MP_ROM_QSTR(MP_QSTR_ppn) },
+  { MP_ROM_QSTR(MP_QSTR_ppn_w), MP_OBJ_NEW_SMALL_INT(screenWidth)},
+  { MP_ROM_QSTR(MP_QSTR_ppn_h), MP_OBJ_NEW_SMALL_INT(screenHeight)},
+  { MP_ROM_QSTR(MP_QSTR_ppn_y0), MP_OBJ_NEW_SMALL_INT(screenY0)},
+  { MP_ROM_QSTR(MP_QSTR_ppn_has_color), hasColor ? mp_const_true : mp_const_false},
```
Comment (RedGl0w) [inline python/port/mod/ppn/modppn_table.c:11]: If this library only run on numworks, why having ternary condition, which will take useless ROM

Comment (RedGl0w) [reply python/port/mod/ppn/modppn_table.c:11]: Globaly, try to use more precompilator directives to avoid useless ROM usage

Comment (AntoninLoubiere) [reply python/port/mod/ppn/modppn_table.c:11]: I did this change in local because, I think it is the cause of the error of compilation.

**Pull request closed without merging**